applications/chain-extension-suite.md

```diff
@@ -0,0 +1,227 @@
+
+
+## Team
+
+### Contact
+
+- **Contact Name:** Jakob Drzazga
+- **Contact Email:** 	admin@deep-ink.ventures
+
+### Legal Structure
+
+Registered Address: Solmsstraße 11 10961 Berlin - Kreuzberg
+Registered Legal Entity: Deep Ink Ventures GmbH
+
+
+- Deep Ink Ventures GmbH, registered with the commercial register at the local court of Berlin, HRB 247342
+
+
+## Development Roadmap
+
+### Overview
+
+- **Total Estimated Duration:** 2-3 month
+- **Full-Time Equivalent (FTE):**  2 FTE
+- **Total Costs:** $ 30,000
+
+
+## Development Status :open_book:
+
+### Milestone 1
+
+- **Estimated Duration:** 3 month
+- **FTE:** 2
+- **Costs:** $30,000
+
+
+| Number | Deliverable          | Specification |
+| ------ | -------------------- | ------------- |
+| **0a.**| License              | Apache 2.0    |
+| **0b.**| Documentation        | Comprehensive documentation covering the CLI Extension tool, its features, and how to use it for generating chain extensions. |
+| **0c.**| Testing Guide        | A detailed guide on how to run the test suites for the CLI Extension tool and a demonstration of the testing coverage. |
+| **1.** | CLI Extension Tool - Part 1: Setup   | Establishing the CLI tool to inspect the Substrate runtime and enable code generation capabilities. |
+| **2.** | CLI Extension Tool - Part 2: Substrate-side Chain Extensions Generation | Auto-generation of the Substrate-side chain extension code based on the inspected runtime methods. |
+| **3.** | CLI Extension Tool - Part 3: ink!-side Chain Extension Traits Generation | Auto-generation of ink!-side chain extension traits to facilitate communication with the generated Substrate-side chain extensions. |
+| **4.** | CLI Extension Tool - Part 4: Reference Smart Contract Generation | Generation of a reference smart contract in ink! that interacts with the generated chain extension traits and Substrate-side chain extensions. |
+| **5.** | Medium Article | A Medium article detailing and announcing the CLI Extension tool, explaining how it simplifies the generation of chain extensions and enhances the developer experience within the Substrate ecosystem. |
+
+## Future Outlook
+If this grant is successfully implemented and received well by the community, we plan to add the following features:
+
+### `pallet_assets_extensions` and `pallet_nft_extensions`
+
+The Substrate framework's FRAME provides `pallet_assets` for fungible tokens and `pallet_nft` for non-fungible tokens (NFTs), aligning with the prevalent token standards in the blockchain domain. On the smart contract side, ink! adheres to PSP-22 for fungible tokens and PSP-34 for NFTs.
+
+Within the objective of this grant we develop `pallet_assets_extensions` and `pallet_nft_extensions`. These extensions aim to expose chain extension functionality for `pallet_assets` and `pallet_nft`, thereby creating a bridge for enhanced interaction between the Substrate runtime and the smart contract environment of ink!.
```

Comment:
```suggestion
Within the objective of this grant we develop `pallet_assets_extensions` and `pallet_nfts_extensions`. These extensions aim to expose chain extension functionality for `pallet_assets` and `pallet_nfts`, thereby creating a bridge for enhanced interaction between the Substrate runtime and the smart contract environment of ink!.
```

Reply:
To be clear, the pallet is called [pallet_nfts](https://docs.rs/pallet-nfts/latest/pallet_nfts/)
For consistency, I think `pallet_nfts_extensions` would be the more appropriate name here